File: Digivance.Auth.Data/Commands/CreateUser.cs
namespace Digivance.Auth.Data.Commands
{
    public record CreateUser
    {
        /// <summary>
        /// Optional display name that the user can choose to identify themselves as
        /// </summary>
        public string DisplayName { get; set; }

        /// <summary>
        /// Required unique email address of this user account
        /// </summary>
        public string EmailAddress { get; set; }

        /// <summary>
        /// Unique id of the tenant that this user account exists in
        /// </summary>
        public Guid TenantId { get; set; }

        /// <summary>
        /// Optional, unique per tenant if provided, custom username of this user account
        /// </summary>
        public string Username { get; set; }
    }
}

File: src/digivance_auth/packages/apiclient/commands/CreateUser.ts
/**
 * Command to create a new user account
 */
export default interface CreateUser {
    /**
     * Optional display name that the user can choose to identify themselves as
     */
    displayName?: string;

    /**
     * Required unique email address of this user account
     */
    emailAddress: string;

    /**
     * Clear text password to set
     */
    password: string;

    /**
     * Unique id of the tenant that this user account exists in (can be null for default)
     */
    tenantId?: string; // Actually a Guid

    /**
     * Optional, unique per tenant if provided, custom username of this user account
     */
    username?: string;
}